### PR TITLE
graph modified

### PR DESCRIPTION
**Pull request by Priyansh19077**
# Summary of changes

# Checklist

- [ ] Checked that there aren't other open pull requests for
  the same change.
- [ ] Added tests for any changes that have a runtime impact.
- [ ] Checked that all tests and type checking passes.
- For changes that have a potential impact on users of this project:
  - [ ] Updated relevant documentation to avoid inaccuracies.
  - [ ] Considered adding additional documentation.
  - [ ] Considered adding an example in `./examples` for new features.
  - [ ] Considered updating our changelog (`CHANGELOG.md`).
- [ ] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

**Comment by Priyansh19077** [comment]
graph modified for recursive depth

File: .gitignore [hunk 1]
```diff
@@ -151,3 +151,5 @@ dmypy.json
 # Cython debug symbols
 cython_debug/
 
+#Remove DS_Store: Mac
+**/.DS_Store
```
Comment (aucampia) [inline .gitignore:155]: This should go in your global gitignore.

File: rdflib/graph.py [hunk 2]
```diff
@@ -347,6 +348,9 @@ def __init__(
         self.context_aware = False
         self.formula_aware = False
         self.default_union = False
+        resource.setrlimit(resource.RLIMIT_STACK, (2 ** 29, -1))
+        sys.setrecursionlimit(10 ** 6)
+
```
Comment (aucampia) [inline rdflib/graph.py:353]: Why can't users of this library not just set the recursion limit as they desire?

**Comment by aucampia** [comment]
@Priyansh19077 if you have time do share the issue you were trying to address with this change, there may be other ways in which we can improve RDFLib to not need a very high recursion limit, and we have made such changes in the past.

**Pull request closed without merging**